### PR TITLE
VTA-973: Implement pagination in checkOpenVisit in activities service

### PR DESCRIPTION
**Pull request by cb-cs**
## VTA-973: Implement pagination in checkOpenVisit in activities service

I have implemented pagination as per getActivities functionality, I have also updated dependencies as per npm audit recommendation. Added additional tests to improve code coverage.
[VTA-973](https://dvsa.atlassian.net/browse/VTA-973)

## Checklist

- [X] Code has been tested manually
- [X] PR title includes the JIRA ticket number
- [X] Branch is rebased against the latest develop
- [X] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/services/DynamoDBService.ts [hunk 1]
```diff
@@ -113,11 +113,11 @@ export class DynamoDBService {
   /**
    * Retrieves the ongoing activity for a given staffId
    * @param staffId - staff id for which to retrieve activity
-   * @returns Promise<PromiseResult<DocumentClient.QueryOutput, AWSError>>
+   * @returns Promise<IActivity[]>
```
Comment (Sul-Aga-kainos) [inline src/services/DynamoDBService.ts:116]: Can you rename IActivity to Activity instead?

File: src/services/DynamoDBService.ts [hunk 2]
```diff
@@ -131,7 +131,13 @@ export class DynamoDBService {
       ExpressionAttributeValues: filterValues
     };
 
-    return DynamoDBService.client.query(query).promise();
+    console.log('params for getOngoingByStaffId', query);
```
Comment (Sul-Aga-kainos) [inline src/services/DynamoDBService.ts:134]: the console log stmt needs to be before return stmt otherwise it will not be executed

Comment (cb-cs) [reply src/services/DynamoDBService.ts:134]: Not sure if the comparison has gone awry, but the log statement is before the call, that return statement has been completely removed.

    console.log('params for getOngoingByStaffId', query);
    try {
      return await this.queryAllData(query);
    } catch (err) {
      console.error('error on getOngoingByStaffId', err);
      throw err;
    }

File: src/services/DynamoDBService.ts (hunk 1 above)
Comment (cb-cs) [reply src/services/DynamoDBService.ts:116]: Updated as per suggestion

**Review by m-coslett — APPROVED** [review]

File: src/services/DynamoDBService.ts [hunk 3]
```diff
@@ -131,7 +131,13 @@ export class DynamoDBService {
...
+    try {
+      return await this.queryAllData(query);
+    } catch (err) {
```
Comment (Daniel-Searle) [inline src/services/DynamoDBService.ts:137]: Can the parameter be renamed to be error

**Review by Daniel-Searle — APPROVED** [review]

**Review by m-coslett — APPROVED** [review]

**Pull request merged**